### PR TITLE
Add components and menu

### PR DESCRIPTION
**Pull request by localconst**

### CODE REVIEW

File: Bubelinez.Tests/test1.cs [hunk 1]
```diff
@@ -8,7 +10,8 @@ public class UnitTest1
         [Fact]
         public void TestReturnToBasic()
         {
-            Assert.Equal(true, true);
+            var result = Intersection.CheckPointRectIntersect(new Vector2f(15, 15), new FloatRect(10, 10, 40, 20));
+            Assert.True(result);
         }
     }
-}
+}
```
Comment (rumyax) [inline Bubelinez.Tests/test1.cs]:
```suggestion
}

```

File: Bubelinez/Components/Button.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        }
+
+        protected Sprite Sprite;
+
+        public virtual void Draw(RenderTarget target, RenderStates states)
+        {
+            target.Draw(Sprite);
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Components/Button.cs]:
```suggestion
}

```

File: Bubelinez/Components/NavigationButton.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            Info = info;
+            Sprite = sprite;
+        }
+
+        public void Draw(RenderTarget target, RenderStates states)
+        {
+            target.Draw(Sprite);
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Components/NavigationButton.cs]:
```suggestion
}

```

File: Bubelinez/EventsArgs/NavigationTriggerArgs.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    public class NavigationTriggerArgs : EventArgs
+    {
+        public LayersEnum Layer { get; private set; }        
+
+        public NavigationTriggerArgs(LayersEnum layer)
+        {
+            Layer = layer;
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/EventsArgs/NavigationTriggerArgs.cs]:
```suggestion
}

```

File: Bubelinez/Interfaces/Layer.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Bubelinez.Interfaces
+{
+    public interface Layer
+    {
+        void HandleMouseEvent(object sender, MouseButtonEventArgs e);
+        event EventHandler<NavigationTriggerArgs> NavigationTriggerArgs;
+        void Draw(RenderTarget target, RenderStates states);
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Interfaces/Layer.cs]:
```suggestion
}

```

File: Bubelinez/Interfaces/LayersEnum.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Bubelinez.Interfaces
+{
+    public enum LayersEnum
+    {
+        None,
+        Menu,
+        Settings,
+        Game
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Interfaces/LayersEnum.cs]:
```suggestion
}

```

File: Bubelinez/Layers/LayerFactory.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+                case LayersEnum.Menu:
+                {
+                    return new Menu();
+                    break;
+                }
+                default: return new Menu();
+            }
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Layers/LayerFactory.cs]:
```suggestion
}

```

File: Bubelinez/Layers/LayerManager.cs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            layer.NavigationTriggerArgs += SwitchLayer;
+        }
+        
+        private void UnsubscribeLayerFromMouseButtonPressed(Layer layer)
+        {
+            _window.MouseButtonPressed -= layer.HandleMouseEvent;
+            layer.NavigationTriggerArgs -= SwitchLayer;
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Layers/LayerManager.cs]:
```suggestion
}

```

File: Bubelinez/Layers/Menu.cs [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        public void Draw(RenderTarget target, RenderStates states)
+        {
+            target.Draw(_background);
+            foreach (var button in _buttons)
+            {
+                target.Draw(button);
+            }
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Layers/Menu.cs]:
```suggestion
}

```

File: Bubelinez/Utils/Intersection.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+        public static bool CheckPointRectIntersect(Vector2f point, FloatRect rect)
+        {
+            if (point.X < rect.Left) return false;
+            if (point.Y < rect.Top) return false;
+            if (point.X > rect.Left + rect.Width) return false;
+            if (point.Y > rect.Top + rect.Height) return false;
+            return true;
+        }
+    }
+}
```
Comment (rumyax) [inline Bubelinez/Utils/Intersection.cs]:
```suggestion
}

```

File: Bubelinez/Window/MainWindow.cs [hunk 11]
```diff
@@ -44,4 +51,4 @@ private void WindowOnClosed(object sender, EventArgs e)
             _window.Close();
         }
     }
-}
+}
```
Comment (rumyax) [inline Bubelinez/Window/MainWindow.cs]:
```suggestion
}

```

File: Bubelinez/Layers/LayerManager.cs [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        public LayerManager(LayerFactory layerFactory, RenderWindow window)
+        {
+            _window = window;
+            _layerFactory = layerFactory;
+            
+            CurrentLayer = layerFactory.getLayer(LayersEnum.Menu);
+            SubscribeLayerToMouseButtonPressed(CurrentLayer);
+        }
+
+        public void Processe()
```
Comment (DenisRumyantsev) [inline Bubelinez/Layers/LayerManager.cs:26]: You have made a bullshit here, comrade!

**Review by JovannyM — APPROVED** [review]
Nice

File: Bubelinez/Layers/LayerFactory.cs [hunk 13]
```diff
@@ -0,0 +1,21 @@
+﻿using Bubelinez.Interfaces;
+
+namespace Bubelinez.Layers
+{
+
```
Comment (rumyax) [inline Bubelinez/Layers/LayerFactory.cs]:
```suggestion
```

**Review by rumyax — APPROVED** [review]
`L` `G` `T` `M` `!`

**Pull request merged**